### PR TITLE
LCAM-1362|Create CRUD end points for USERS table

### PR DESCRIPTION
**Pull request by venkat980**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1362)

Added CRUD end points for USERS table.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/test/java/gov/uk/courtdata/users/controller/UserSummaryControllerTest.java [hunk 1]
```diff
@@ -33,13 +35,79 @@ void givenIncorrectParameters_whenUpdateCCOutcomeIsInvoked_thenErrorIsThrown() t
...
                 .andExpect(status().isOk())
                 .andExpect(content().contentType(MediaType.APPLICATION_JSON))
-                .andExpect(jsonPath("$.username").value(TestModelDataBuilder.TEST_USER));
+                .andExpect(jsonPath("$.username").value(TEST_USER));
     }
 
+    @Test
+    void givenIncorrectParameters_whenPatchUserIsInvoked_thenErrorIsThrown() throws Exception {
+        mvc.perform(MockMvcRequestBuilders.post(USER_ENDPOINT_URL)
+                        .contentType(MediaType.APPLICATION_JSON))
```
Comment (skasthuri36326) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/users/controller/UserSummaryControllerTest.java:81]: For patch, you need to change to MockMvcRequestBuilders.patch

File: maat-court-data-api/src/main/java/gov/uk/courtdata/users/service/UserSummaryService.java [hunk 2]
```diff
@@ -55,4 +58,33 @@ public UserSummaryDTO getUserSummary(String username) {
...
+    }
+
+    public void deleteUser(String username) {
+        userRepository.deleteById(username);
+    }
+
+    public void patchUser(String username, Map<String, Object> updateFields) {
+        UserEntity userEntity = getUser(username);
+        updateFields.forEach((key, value) -> {
+            Field field = ReflectionUtils.findField(UserEntity.class, key);
```
Comment (skasthuri36326) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/users/service/UserSummaryService.java:82]: As there is less attributes in the UserEntity, I prefer put over PATCH

File: maat-court-data-api/src/test/java/gov/uk/courtdata/users/controller/UserSummaryControllerTest.java (hunk 1 above)
Comment (venkat980) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/users/controller/UserSummaryControllerTest.java:81]: Fixed.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/users/service/UserSummaryService.java (hunk 2 above)
Comment (venkat980) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/users/service/UserSummaryService.java:82]: Keeping it consistent with the recent changes, like Reservations that is used only for the automated tests.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/users/controller/UserSummaryController.java [hunk 3]
```diff
@@ -38,4 +41,39 @@ public ResponseEntity<UserSummaryDTO> getUserSummary(
...
+    @StandardApiResponse
+    @ApiResponse(responseCode = "200", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE))
+    public ResponseEntity<Void> createUser(@RequestBody UserEntity userEntity) {
+        userSummaryService.createUser(userEntity);
+        return ResponseEntity.ok().build();
+    }
+
+    @PatchMapping("/{username}")
+    @Operation(description = "Update user details")
+    @StandardApiResponseCodes
```
Comment (nigelpain) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/users/controller/UserSummaryController.java:64]: Do we need the `@NotFoundApiResponse` annotation here aswell, as the patch service calls the get service to find the user to update which throws a `RequestedObjectNotFoundException` if one can't be found?

**Review by skasthuri36326 — APPROVED** [review]

File: maat-court-data-api/src/main/java/gov/uk/courtdata/users/controller/UserSummaryController.java (hunk 3 above)
Comment (venkat980) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/users/controller/UserSummaryController.java:64]: Good spot. Added the annotation.

**Pull request merged**